src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -103,6 +136,34 @@ def __init__(
                 tupleize_cols=tupleize_cols,
             )
 
+    def __getattr__(self, key: str) -> Any:
+        """
+        Return item identified by `key`.
+
+        Parameters
+        ----------
+        key : str
+            Key to get.
+
+        Returns
+        -------
+        Any
+
+        Notes
+        -----
+        This method also helps raise NotImplementedError for APIs out
+        of current scope that are not implemented.
+        """
+        # TODO: SNOW-1063347: Modin upgrade - modin.pandas.Series functions
+        try:
+            return object.__getattribute__(self, key)
+        except AttributeError as err:
+            if key in INDEX_METHODS_NOT_IMPLEMENTED:
```

Comment:
You don't need this list, you can create a pandas.Index([]) object and check whether the key is an attribute of it, If it is, then raise NotImplementedError. Does that make sense?

Reply:
Makes sense but would that not cause code coverage issues if I do it that way without getting rid of all the methods below? I can get rid of them if you think that's better

Reply:
If the methods is in our plan, we can keep them. Otherwise, we can remove them.